### DISCUSSION TITLE
Kurdish language support

### DISCUSSION THREAD

**Discussion by luiscarlos2000** [RHVoice/RHVoice#660] [Ideas]
Hi,
First time I discuss something here on the RHVoice discussion, as I sometimes use RHVoice.
I'm not a native kurdish language speaker, but recently someone asked to implement that language, as them has some voice data to create them.
If possible, will RHVoice support Kurdish? No matter the variety or only as a single language.
Best,
Luis.

**Comment by CrowdsourceKurdish** [comment]
Hi @Olga-Yakovleva @alex19EP
We have a language community, our community is creating open source datasets for the Kurdish language, and we currently have a 64-hour Kurdish dataset(Published).
The new dataset, which will be released next month, will include more than 100 hours of voice datasets.
Can you train RHVoice with this dataset?
Source for Dataset;
https://commonvoice.mozilla.org/kmr/datasets

We will be very happy to see the Kurdish language in RHVoice.

Best Regards
Kurdish Crowdsource Team

**Comment by zstanecic** [comment]
Hi
@CrowdsourceKurdish,
About which kurdish variety are we talking about? Sorani or kurmanji?
Firstly, you will need to write rules for kurdish and create the language module,
see the polish language as an example.
To write rules, you need to master the foma language. 
Google for the foma regex reference, you will find it.

**Reply by CrowdsourceKurdish** [reply]
Hello, 
I mean (kurmanji-ku/kmr), but we can also do it for other Kurdish dialects. 120 hours of collective voice data is currently available for Kurdish (sorani-ckb).
I am fluent in the Kurdish language, but I am not good at coding and creating language modules. I'm going to talk to a friend who is interested in audio technologies to create a language module.
We have a 60-hour dataset created with the voices of 500 people, which will be more than 100 lines when the new version is released next month.
Thanks.

**Comment by Olga-Yakovleva** [comment]
Greetings everyone! As the previous comment shows, developing support for a new language in RHVoice is a rather involved task. There are two development models our voices follow: funded voices and community voices. Ann example of the first are the Macedonian and Albanian voices developed last year (see https://louderpages.org). Ann example of the second are the Polish voices developed by @grzezlo and @zstanecic. The second path needs one or more developers able and willing to dedicate time and effort to the development process. One of the resources they would need is a speech corpus. Unfortunately, RHVoice doesn't implement support for building voices from multispeaker corpora such as the one linked to here. We are still using the old way of recording enough speech from a professional speaker in a high quality recording environment. I'm sorry not to be able to give easy answers. I understand that this is one of the languages truely needing a TTS voice and overlooked by the large players. But this is never simple for any language I'm afraid.

**Reply by CrowdsourceKurdish** [reply]
Hello,
I understand, thank you for the information.
How many hours of audio corpus from a single speaker is needed?

**Reply by louderpages** [reply]
Hi @CrowdsourceKurdish - With audio it is not a simple matter of hours.  The audio must cover a set of text  - a corpus - which has been designed to train a voice with sufficient phonetic coverage.    In order to design such a corpus, you first need to build the language module.  Also, in order to do anything with the audio, you must supply a language module to the training software.

Developing a language module is a hard task even for an easy language which is written phonemically.  The software tool used - FOMA for Finite State Transducers - is a very peculiar technology which takes a lot of time to learn. And even longer to teach.  This is the problem.

**Reply by CrowdsourceKurdish** [reply]
Hi, @Olga-Yakovleva @zstanecic @louderpages 
I know of an open-source Kurdish (kurmanji) model trained for Stanza, but I don't know if it can be used for RHvoice. Detailed information about the Kurdish language is also available below.
1-https://github.com/UniversalDependencies/UD_Kurmanji-MG
2-https://zenodo.org/record/6558295#.Y1lyF7bP3rc
3-https://universaldependencies.org/treebanks/kmr_mg/index.html

**Comment by zstanecic** [comment]
Hi Olga and all,

I have a python script which can process and sort the large dataset mentioned in the discussion. The script is also able to generate ssml from it.

Kurdish, especially the kurmanji variation is written in latin. Ignore Sorani, as it is written in Arabic script, which is rather tough for processing.

The dataset mentioned is for kmr, that is the Kurdish spoken in Rojawa, Turkey and parts of Iraq.

Best,

Zvonimir

**Comment by CrowdsourceKurdish** [comment]
Hello,
Yes, Kurdish (kurmanji- ku/kmr) uses the Latin alphabet. Right now, our community is creating collective voice data, but we're thinking about creating professional voice data from a single speaker in the near future.
What are the minimum hours of audio data needed for a good result?

**Reply by louderpages** [reply]
Hi  @CrowdsourceKurdish.  We have been receiving requests from Iraqi Kurdistan to build a voice.  But communication in English is difficult.  Can you help?  We would be interested in Latin-based Kurdish.  One of recently at home town in Scotland with our Turkmen voice talent when we bought some french fries from a Kurdish kebab restaurant.  It is time. :)  Please email privately to inq at louderpages dot org
